networkit/cpp/embedding/AliasSampler.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * AliasSampler.cpp
+ *
+ *  Created on: 03.07.2020
+ *      Author: Klaus Ahrens  <ahrens@informatik.hu-berlin.de>
+ *
+ *  adapted and reimplemented from node2vec
+ *  part of snap [https://github.com/snap-stanford/snap]
+ *  Copyright (c) 2007-2019, Jure Leskovec (under BSD license)
+ *
+ *  see [https://arxiv.org/pdf/1607.00653v1.pdf]
+ *
+ */
+
+// networkit-format
+
+#include <random>
+#include <vector>
+
+#include <networkit/Globals.hpp>
+#include <networkit/auxiliary/Random.hpp>
+
+#include "AliasSampler.hpp"
+
+namespace NetworKit {
+namespace Embedding {
+
+double uniform_real() {
+    static std::uniform_real_distribution<> dist;
+
+    return dist(Aux::Random::getURNG());
+}
+
+void AliasSampler::unigram(std::vector<float> &probs) {
+    auto N = probs.size();
+
+    std::vector<index> underV;
+    std::vector<index> overV;
+    for (index i = 0; i < N; ++i) {
+        U[i] = probs[i] * N;
+        if (U[i] < 1) {
+            underV.push_back(i);
+        } else {
+            overV.push_back(i);
+        }
+    }
+    while (!underV.empty() && !overV.empty()) {
+        auto small = underV.back();
+        auto large = overV.back();
+        underV.pop_back();
+        overV.pop_back();
+        K[small] = large;
+        U[large] = (U[large] + U[small]) - 1;
+        if (U[large] < 1) {
+            underV.push_back(large);
+        } else {
+            overV.push_back(large);
+        }
+    }
+    while (!underV.empty()) {
+        auto curr = underV.back();
+        underV.pop_back();
+        U[curr] = 1;
+    }
+    while (!overV.empty()) {
+        auto curr = overV.back();
+        overV.pop_back();
+        U[curr] = 1;
+    }
```

Comment:
These two loops can be simplified with
```
for (index i : underV/overV)
    U[i] = 1;
```
The two vectors will go out of scope anyway so there is no need to `pop_back` elements right?.